### PR TITLE
docs: Add new telemetry to changelog

### PR DESCRIPTION
**Pull request by Dan-Heath**
Adds the telemetry work done in ICU-13864 to the changelog for release 0.16.1.

### CODE REVIEW

File: CHANGELOG.md [hunk 1]
```diff
@@ -4,6 +4,11 @@ Canonical reference for changes, improvements, and bugfixes for Boundary.
 
 ## Next
 
+### New and Improved
+
+* The observation tag was added to `storage_bucket.proto` and `storage_bucket_service_proto` for telemetry purposes. If you enable telemetry and observation events, Boundary will now collect data about session recording and storage buckets.
```
Comment (sepehrfrgh) [inline CHANGELOG.md:9]: My suggestion would be to combined storage bucket and session recording together into one sentence like below:
```suggestion
* The observation tag was added to Session Recording and Storage Bucket proto messages for telemetry purposes. If you enable telemetry and observation events, Boundary will now collect data about session recording and storage buckets.
```

File: CHANGELOG.md [hunk 2]
```diff
@@ -4,6 +4,11 @@ Canonical reference for changes, improvements, and bugfixes for Boundary.
...
+([PR](https://github.com/hashicorp/boundary/pull/4824))
```
Comment (sepehrfrgh) [inline CHANGELOG.md:10]: PR for session recording https://github.com/hashicorp/boundary/pull/4825

File: CHANGELOG.md (hunk 1 above)
Comment (Dan-Heath) [reply CHANGELOG.md]: This sounds great and is easier for the user to understand. 👍

**Review by sepehrfrgh — APPROVED** [review]
LGTM thank you Dan 👍

**Pull request merged**